modules/gtksourceview/overlay.nix

```
@@ -0,0 +1,38 @@
+{ config, lib, ... }:
+
+let
+  inherit (lib) mkEnableOption optionalAttrs;
+
+  cfg = config.stylix.targets.gtksourceview;
+
+  style = config.lib.stylix.colors {
+    template = ./template.xml.mustache;
+    extension = "xml";
+  };
+
+  attrsOverride = version: oldAttrs: {
+    postFixup = ''
+      ${oldAttrs.postFixup or ""}
+      styles_dir="$out/share/gtksourceview-${version}/styles"
+      mkdir --parents "$styles_dir"
+      cp ${style} "$styles_dir/stylix.xml"
+    '';
+  };
+in
+{
+  options.stylix.targets.gtksourceview = {
+    enable = config.lib.stylix.mkEnableTarget "GTKSourceView" (config ? home);
+    overlay.enable = mkEnableOption "the overlay";
+  };
```

shouldn't it be this?
```suggestion
  options.stylix.targets.gtksourceview = with config.lib.stylix; {
    enable = mkEnableTarget "GTKSourceView" true;
    overlay.enable = mkEnableTarget "the overlay" (! config ? home);
  };
```

---

Yea right, allthough I'm not sure whether mkEnableTarget should be used for this, because the overlay isn't strictly a target.

---

mkEnableTarget is just a wrapper over mkEnableOption:
```nix
humanName: autoEnable:
lib.mkEnableOption "theming for ${humanName}"
// {
  default = cfg.enable && cfg.autoEnable && autoEnable;
  example = !autoEnable;
}
// lib.optionalAttrs autoEnable {
  defaultText = lib.literalMD "same as `stylix.autoEnable`";
};
```

---

hmmm actually your right it shouldn't be used

---

it should be:
```
overlay.enable = mkEnableTarget "the overlay" // { default = !(config ? home); };